resources/file.rb

```diff
@@ -5,12 +5,27 @@
 # Author:: Jamie Winsor (<jamie@vialstudios.com>)
 #
 
-actions :extract
-default_action :extract
+property :path, kind_of: String, name_property: true, required: true
+property :owner, kind_of: String
+property :group, kind_of: String
+property :mode, kind_of: Integer, default: 0755
+property :extract_to, kind_of: String, required: true
+property :extract_options, kind_of: [Array, Symbol], default: []
 
-attribute :path, kind_of: String, name_attribute: true, required: true
-attribute :owner, kind_of: String
-attribute :group, kind_of: String
-attribute :mode, kind_of: Integer, default: 0755
-attribute :extract_to, kind_of: String, required: true
-attribute :extract_options, kind_of: [Array, Symbol], default: []
+require 'fileutils'
+
+action :extract do
+  unless ::File.exist?(new_resource.path)
+    raise Errno::ENOENT, "no archive found at #{new_resource.path}"
+  end
+
+  FileUtils.mkdir_p(new_resource.extract_to, mode: new_resource.mode)
```

Comment:
The actual work that occurs here needs to be wrapped in a converge_by block so that this doesn't blow up under why-run mode. It will also give users a nicer converge experience.